### PR TITLE
Add additional code validation

### LINKED ISSUE #24: Calculate check character on diga code before validating against the API
## Problem

In the documentation at gkv and bitmarck's FAQ there is some mention of the manufacturer having to check the code is valid before sending it to the api (?). They are likely referring to the check character (last character) of the code. This does not seem urgent to implement as the API will respond if this is incorrect anyway, but it would be a nice to have feature.

## Solution

In [DigaCodeDefaultParser class](https://github.com/alex-therapeutics/diga-api-client/blob/4ee54af78aa512ea3a370d4c3ffdebdd060df4e3/src/main/java/com/alextherapeutics/diga/implementation/DigaCodeDefaultParser.java#L84) there is a method for validating the code

It seems bitmarck is referring to https://base32check.org/ for doing this. Try to implement that (with the check character specification at gkv) and add it to the code validation method.

### PR DESCRIPTION
**Pull request by gtuk**
# Pull Request

### Description
Adds an additional code validation based on https://github.com/bitmarck-service/base32check-java

### Closes
#24

### CODE REVIEW

File: src/main/java/com/alextherapeutics/diga/implementation/DigaCodeDefaultParser.java [hunk 1]
```diff
@@ -93,9 +94,23 @@ private ParsedDigaCode parseCodeString(String codeString) {
   }
 
   private boolean validateDigaCodeStructure(String code) {
-    // TODO - make more check, like regex, the different parts, etc.
```
Comment (SebastianRehfeldt) [inline src/main/java/com/alextherapeutics/diga/implementation/DigaCodeDefaultParser.java:96]: If I remember the documentation correctly the codes can consist of `[A-Z]` or `[2-7]`. How does the base32check and diga api behave if codes contain lower-cased characters, e.g. by typos when entering prescription codes? Would it be a good idea to always capitalize the codes within the diga-api-client?

Comment (gtuk) [reply src/main/java/com/alextherapeutics/diga/implementation/DigaCodeDefaultParser.java:96]: I would leave it to the client implementation to send the proper (uppercase) code

**Comment by fongie** [comment]
thanks @gtuk ! It looks good at first glance, but I will want to test run the code and since this is not a critical bugfix and I'm very busy right now it might take until next week for me to review this. Just so you know 😃

File: src/main/java/com/alextherapeutics/diga/implementation/DigaCodeDefaultParser.java (hunk 1 above)
Comment (SebastianRehfeldt) [reply src/main/java/com/alextherapeutics/diga/implementation/DigaCodeDefaultParser.java:96]: Fair enough 👍

**Comment by gtuk** [comment]
@fongie don't worry, as you said. It's nothing urgent to be merged

**Review by fongie — APPROVED** [review]
thanks!

**Comment by fongie** [comment]
live in 1.2.0 @gtuk 😃

**Pull request merged**